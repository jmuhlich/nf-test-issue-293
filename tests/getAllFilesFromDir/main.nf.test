nextflow_pipeline {

    name "Test getAllFilesFromDir"
    script "./main.nf"
    tag "getAllFilesFromDir"

    test("getAllFilesFromDir") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
          // Use getAllFilesFromDir() to get a list of all files and folders from the output directory, minus the timestamped files
          def stable_name       = getAllFilesFromDir(params.outdir, true, ['pipeline_info/execution_*.{html,txt}'], null, ['**'])
          // Use getAllFilesFromDir() to get a list of all files and folders from the output directory, minus the timestamped files
          def stable_name_again = getAllFilesFromDir(params.outdir, true, [''], null, ['stable/*'])
          // Use getAllFilesFromDir() to get a list of all files from the output directory, minus the non-stable files
          def stable_content    = getAllFilesFromDir(params.outdir, false, ['pipeline_info/execution_*.{html,txt}'], 'tests/getAllFilesFromDir/.nftignore', ['**'])
          assert snapshot(
            // Only snapshot name
            stable_name*.name,
            // Only snapshot name again
            stable_name_again*.name,
            // Snapshot content
            stable_content,
            // Capture file with includeGlobs
          ).match()
        }
    }

}
