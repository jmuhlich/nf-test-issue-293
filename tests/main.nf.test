nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"

    test("first test") {
        then {
            assert snapshot([]).match()
        }
    }

}
