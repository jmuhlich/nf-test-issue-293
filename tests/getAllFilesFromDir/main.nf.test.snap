{
    "getAllFilesFromDir with named params": {
        "content": [
            [
                "pipeline_info",
                "stable",
                "stable_content.txt",
                "stable_name.txt"
            ],
            [
                "stable_content.txt",
                "stable_name.txt"
            ],
            [
                "stable_content.txt:md5,f6d73f703cda1c725ea78369a6c3a48d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.09.0"
        },
        "timestamp": "2024-10-07T16:53:04.994417"
    },
    "getAllFilesFromDir": {
        "content": [
            [
                "pipeline_info",
                "stable",
                "stable_content.txt",
                "stable_name.txt"
            ],
            [
                "stable_content.txt",
                "stable_name.txt"
            ],
            [
                "stable_content.txt:md5,f6d73f703cda1c725ea78369a6c3a48d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.09.0"
        },
        "timestamp": "2024-10-07T16:53:01.146706"
    }
}
